### DISCUSSION TITLE
3 UART question (2 UART + USB serial)

### DISCUSSION THREAD

**Discussion by pike2k** [earlephilhower/arduino-pico#2074] [General]
I need 2 UARTs in my project so I want to use Software Serial (over PIO) for USB (serial)debugging. 
I realize that UART0's default pins are GP0 and GP1 and I guess that these are the same pins used for the USB port (Serial)? 

Is the following scenario possible:
UART0 RX/TX moved from default pins 0,1 to either 16,17 or 21,22
UART1 RX/TX on pins 6,7 or 11,12
SoftwareSerial on pins 0,1 (or any other pins as long as I can get output to show over USB)

If my logic is incorrect, I don't mind better suggestions :)
Many thanks in advance

**Comment by JoeEarlam** [comment]
The USB serial is separate from the hardware UARTs, so you can use hardware on the default pins and USB all independently of each other.

**Reply by pike2k** [reply]
Thanks for your reply. This would indeed simplify things a lot.

So you're saying that I can use GP0 and GP1 for what I want (like UART0), while USB serial (Serial. right?) does not use hardware GP pins? While at the same time using Serial1. and Serial2.

**Reply by JoeEarlam** [reply]
Correct!
